integration-libs/punchout/core/services/punchout-auth.service.ts

```diff
@@ -47,19 +57,36 @@ export class PunchoutAuthService {
     );
   }
 
-  loginWithToken(accessToken: string, userId: string): void {
+  loginWithToken(accessToken: string): void {
     // Code mostly based on auth lib we use and the way it handles token properties
     this.authStorageService.setItem(ACCESS_TOKEN, accessToken);
     this.authStorageService.setItem(
       ACCESS_TOKEN_STORED_AT,
       Date.now().toString()
     );
-    this.userIdService.setUserId(userId);
+    this.authStorageService.setItem(TOKEN_TYPE, 'Bearer');
+    this.userIdService.setUserId(OCC_USER_ID_CURRENT);
     this.store.dispatch(new AuthActions.Login());
     this.globalMessageService.remove(GlobalMessageType.MSG_TYPE_CONFIRMATION);
   }
 
   isUserLoggedIn(): Observable<boolean> {
     return this.authService.isUserLoggedIn().pipe(take(1));
   }
+
+  endPunchoutSession(): void {
+    this.punchoutStoreService.clearPunchoutState();
+    this.globalMessageService.add(
+      { key: 'httpHandlers.unauthorized.common' },
+      GlobalMessageType.MSG_TYPE_ERROR
+    );
+
+    if (this.punchoutDetectionService.isPunchoutSessionPage()) {
```

Comment:
we don't want the user to be redirected to punchout page after a manual login, user would get back to login page again, stuck in infinite loop until user naviagte somewhere else.